bitnami/influxdb/templates/influxdb/cronjob-backup.yaml

```diff
@@ -0,0 +1,95 @@
+{{- if .Values.backup.enabled }}
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: "{{ template "influxdb.fullname" . }}-backup"
+  labels:
+    {{- include "influxdb.labels" . | nindent 4 }}
+  annotations:
+    {{- toYaml .Values.backup.annotations | nindent 4 }}
+spec:
+  schedule: {{.Values.backup.schedule | quote }}
+  concurrencyPolicy: Forbid
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels: 
+            {{- include "influxdb.matchLabels" . | nindent 12 }}
+        spec:
+          securityContext:
+            runAsUser: 0
+            runAsGroup: 0
```

Comment:
You should try to use `non-root` containers by default. It's mandatory for K8s distributions such as OpenShift or Tanzu Kubernetes Grid.

Reply:
the find command has a nonzero exitcode on the "lost+found" directory if run as non root.

Reply:
You can skip that:

```diff
- find {{ .Values.backup.directory }}  -mindepth 2 -maxdepth 2 -type d -mtime +{{ .Values.backup.retentionDays }} -exec rm -r {} \;
+ find {{ .Values.backup.directory }} -not -name ".snapshot" -not -name "lost+found" -mindepth 2 -maxdepth 2 -type d -mtime +{{ .Values.backup.retentionDays }} -exec rm -r {} \;
```